commons-test/src/main/java/org/infinispan/commons/test/security/TestCertificates.java

```diff
@@ -0,0 +1,170 @@
+package org.infinispan.commons.test.security;
+
+import java.io.OutputStream;
+import java.io.Writer;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+import javax.security.auth.x500.X500Principal;
+
+import org.wildfly.security.x500.cert.BasicConstraintsExtension;
+import org.wildfly.security.x500.cert.SelfSignedX509CertificateAndSigningKey;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+/**
+ * @since 15.0
+ **/
+public class TestCertificates {
+   public static final String BASE_DN = "CN=%s,OU=Infinispan,O=JBoss,L=Red Hat";
+   public static final char[] KEY_PASSWORD = "secret".toCharArray();
+   public static final String KEY_ALGORITHM = "RSA";
+   public static final String KEY_SIGNATURE_ALGORITHM = "SHA256withRSA";
+
+   public static final String KEYSTORE_TYPE = KeyStore.getDefaultType();
+   public static final String EXTENSION = ".pfx";
+
+   private static final AtomicLong CERT_SERIAL = new AtomicLong(1);
+
+   public static String certificate(String name) {
+      return baseDir().resolve(name + EXTENSION).toString();
+   }
+
+   public static String pem(String name) {
+      return baseDir().resolve(name + ".pem").toString();
+   }
+
+   public static void createKeyStores() {
```

Comment:
shouldn't this be invoked when `TestCertificates` is loaded instead of being done by the tests?

Reply:
moving it to a static initializer